### PR TITLE
ct: fix bugs for the demo

### PR DESCRIPTION
**Pull request by danhhz**
Adapts my demo script to be an slt test and fixes the bug found along the way.

Closes https://github.com/MaterializeInc/database-issues/issues/8697

### Motivation

  * This PR fixes a recognized bug.

### Tips for reviewer

There's a tiny bit of polish left to be done here, but I have a hard stop today and wanted to get a CI run in. Feel free to take a look though!

### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered. ([trigger-ci for additional test/nightly runs](https://trigger-ci.dev.materialize.com/))
- [ ] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [ ] If this PR includes major [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note), I have pinged the relevant PM to schedule a changelog post.

### CODE REVIEW

**Review by ParkMyCar — APPROVED** [review]

File: src/adapter/src/coord/sequencer/inner/create_continual_task.rs [hunk 1]
```diff
@@ -278,6 +279,15 @@ fn update_create_sql(
     let f =
         |x: &mut RawItemName| *x = RawItemName::Name(PartialItemName::from(ct_name.clone()).into());
 
+    struct ReplaceName(FullItemName);
+    impl<'ast> VisitMut<'ast, Raw> for ReplaceName {
+        fn visit_item_name_mut(&mut self, node: &'ast mut RawItemName) {
+            if node.name().0.last().map(|x| x.as_str()) == Some(&self.0.item) {
+                *node = RawItemName::Name(PartialItemName::from(self.0.clone()).into());
+            }
```
[the remark this answers is not in the document]
Comment (danhhz) [reply src/adapter/src/coord/sequencer/inner/create_continual_task.rs:287]: Ah hmm, good call. Any ideas for how best to write this instead? We've got a FullItemName and we need to replace any matching partial ones with it. Essentially name resolution, but we haven't written it to the catalog yet

[the remark this answers is not in the document]
Comment (danhhz) [reply src/adapter/src/coord/sequencer/inner/create_continual_task.rs]: @ParkMyCar How's this?

[the remark this answers is not in the document]
Comment (danhhz) [reply src/adapter/src/coord/sequencer/inner/create_continual_task.rs]: CI and nightlies seems happy with it, so I'll go ahead and merge to get this in, but happy to fix anything up post-merge

**Review by danhhz — COMMENTED** [review]
TFTR!

**Pull request merged**